scripts/github/migrate_to_external_accounts.py

```diff
@@ -0,0 +1,377 @@
+import logging
+import os
+import sys
+import urlparse
+
+from nose.tools import *  # noqa
+from modularodm import Q
+
+from framework.mongo import database
+from framework.transactions.context import TokuTransaction
+
+from website import settings
+from website.app import init_app
+from website.models import User, Node
+from website.oauth.models import ExternalAccount
+from website.addons.github.api import GitHubClient
+from website.addons.github import settings as github_settings
+from website.addons.github.utils import make_hook_secret
+from website.addons.github.exceptions import GitHubError, ApiError
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+PROVIDER = 'github'
+PROVIDER_NAME = 'GitHub'
+HOOK_DOMAIN = github_settings.HOOK_DOMAIN or settings.DOMAIN
+
+# set of {ExternalAccount._id: (user_settings, oauth_settings)} mappings
+# with invalid credentials, for logging purposes
+invalid_oauth_creds = {}
+
+def verify_user_and_oauth_settings_documents(user_document, oauth_document):
+    try:
+        assert_in('_id', user_document)
+        assert_in('oauth_settings', user_document)
+        assert_in('deleted', user_document)
+        assert_in('owner', user_document)
+        assert_in('_id', oauth_document)
+        assert_in('github_user_id', oauth_document)
+        assert_in('github_user_name', oauth_document)
+        assert_in('oauth_access_token', oauth_document)
+        assert_is_not_none(user_document['owner'])
+        assert_equal(user_document['oauth_settings'], oauth_document['github_user_id'])
+    except AssertionError:
+        return False
+    else:
+        return True
+
+def verify_node_settings_document(document, account):
+    assert_in('_id', document)
+    assert_in('deleted', document)
+    assert_in('hook_id', document)
+    assert_in('repo', document)
+    assert_in('user', document)
+    assert_in('registration_data', document)
+    assert_in('owner', document)
+    assert_is_not_none(document['owner'])
+    assert_in('user_settings', document)
+    try:
+        assert_in('hook_secret', document)
+    except AssertionError:
+        try:
+            add_hook_to_old_node_settings(document, account)
```

Comment:
Could use some logging here to indicate that a request to the GH API is being sent